src/Verify.MSTest.SourceGenerator/Parser.cs

```diff
@@ -1,12 +1,63 @@
+using System.Data;
+using PropertyFlags = ClassToGenerate.PropertyFlags;
+
 static class Parser
 {
     public static ClassToGenerate? Parse(INamedTypeSymbol typeSymbol, TypeDeclarationSyntax typeSyntax, Cancel cancel)
     {
         var ns = typeSymbol.GetNamespaceOrDefault();
         var name = typeSyntax.GetTypeNameWithGenericParameters();
         var parents = GetParentClasses(typeSyntax, cancel);
+        var testContextPropertyFlags =
+            GetDerivedPropertyFlagsGiven(
+                BaseTestContextProperties(typeSymbol).FirstOrDefault());
+
+        return new ClassToGenerate(
+            Namespace: ns,
+            ClassName: name,
+            TestContextPropertyFlags: testContextPropertyFlags,
+            ParentClasses: parents);
+    }
+
+    private static PropertyFlags GetDerivedPropertyFlagsGiven(
+        IPropertySymbol? baseClassProperty)
+    {
+        var isAbstract = baseClassProperty?.IsAbstract;
+        switch (isAbstract)
+        {
+            case true:
+                return PropertyFlags.Override;
+            case false:
+                return PropertyFlags.Override | PropertyFlags.CallBase;
+            case null:
+                return PropertyFlags.None;
+        }
+    }
+
+    static IEnumerable<IPropertySymbol> BaseTestContextProperties(
+        INamedTypeSymbol typeSymbol)
+    =>
+            from baseClass in BaseClassesOf(typeSymbol)
```

Comment:
nit: I think this is the only place in the codebase to use the query-style syntax. It's up to @SimonCropp but I'd assume you want method-style for consistency.